### PR TITLE
feat(swc/plugin_runner): implement initial load for wasm32

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

This was unexpectedly bit tricky due to small bumps across various places. Anyway, this PR covers basic `load` execution paths for the plugin with native runtime (wasm host). It still uses current experimental `resolver` approach which accepts redundant bytes copy between js -> wasm. 

Also, there are couple of issues blocks actual plugin executions yet: 
- wasi binary init fails
- transformexecutor need to be updated to work with wasmer::js
- string-cache upstream fix needed

**Related issue (if exists):**
- https://github.com/swc-project/swc/issues/3934

### CODE REVIEW

File: crates/swc/src/config/mod.rs [hunk 1]
```diff
@@ -486,7 +486,8 @@ impl Options {
                 env_name: self.env_name.to_owned(),
             };
 
-            crate::plugin::plugins(experimental, plugin_context)
+            swc_plugin_runner::cache::init_plugin_module_cache_once();
```
Comment (kdy1) [inline crates/swc/src/config/mod.rs:489]: I think we should check if `plugins` is not empty

Comment (kdy1) [reply crates/swc/src/config/mod.rs:489]: Otherwise, we create a directory named `.swc` even though the user does not use plugin.

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/swc/src/config/mod.rs (hunk 1 above)
Comment (kwonoj) [reply crates/swc/src/config/mod.rs:489]: this is for the wasm target, where `init_plugin_module_cache_once` does not have access to filesystem and its implementation actually do not create any file - only creates hashmap for in memory cache.

Comment (kdy1) [reply crates/swc/src/config/mod.rs:489]: Oh, I see. Thanks!

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_plugin_runner
 - swc

**Pull request merged**